packages/playwright-core/src/server/fetch.ts

```diff
@@ -207,7 +207,7 @@ export abstract class APIRequestContext extends SdkObject {
       if (!cookie.domain)
         cookie.domain = url.hostname;
       else
-        assert(cookie.domain.startsWith('.'));
+        assert(cookie.domain.startsWith('.') || !cookie.domain.includes('.'));
```

Comment:
Is this a user-facing error? Let's do a throw with a meaningful message instead?

Reply:
No, it's not user facing, it's a CHECK.